ToneMapper/TMTonalFeature.m

```diff
@@ -33,19 +52,32 @@ @interface TMTonalFeature()
 /// \c UISegmentedControl object used to select filter operations to alter.
 @property (strong, nonatomic) UISegmentedControl *operatorOptions;
 
+/// \c UISegmentedControl object used to select advanced filter operations to alter.
+@property (strong, nonatomic) UISegmentedControl *advancedContrastOperatorOptions;
+
 @end
 
 @implementation TMTonalFeature
 
 @synthesize delegate = _delegate;
 
-- (instancetype)initWithDrawer:(TMGLDrawer *)drawer {
-  if (!drawer) {
+- (instancetype)initWithDrawer:(TMGLDrawer *)drawer andInputTexture:(TMGLTexture *)inputTexture {
+  if (!drawer || !inputTexture) {
     return nil;
   }
   
   if (self = [super init]) {
-    _filter = [[TMGLToneAdjustFilter alloc] initWithDrawer:drawer];
+      _inputTexture = inputTexture;
+      _bilateralFilter = [[TMGLBilateralFilter alloc] initWithDrawer:drawer];
+      self.bilateralFilter.kernelSize = kSmoothKernelSize;
+      TMGLTexture *smoothTexture = [self.bilateralFilter applyOnTexture:inputTexture];
+      self.bilateralFilter.kernelSize = kSmoothKernelSize;
```

Comment:
Ah now I see what you mean. Previously it was 
```
self.bilateralFilter.kernelSize = kSmoothKernelSize;
TMGLTexture *smoothTexture = [self.bilateralFilter applyOnTexture:inputTexture];
self.bilateralFilter.kernelSize = kSmootherKernelSize;
TMGLTexture *smootherTexture = [self.bilateralFilter applyOnTexture:inputTexture];
```

And when I changed it to `TMGLTexture *smootherTexture = [self.bilateralFilter applyOnTexture:smoothTexture]`
I just changed it instead of removing it.

I'll fix it on monday.